content/ai/concepts/neural-networks/terms/stochastic-gradient-descent/stochastic-gradient-descent.md

```diff
@@ -0,0 +1,61 @@
+---
+Title: 'Stochastic Gradient Desent'
+Description: 'Stochastic Gradient Desent is optimizer algorithm that minimizes the loss functions in machine learning and deep learning models.'
+Subjects:
+  - 'Machine Learning'
+  - 'Deep Learning'
+  - 'Computer Science'
+Tags:
+  - 'AI'
+  - 'Neural Network'
+  - 'Optimizer'
+CatalogContent:
+  - 'paths/computer-science'
+  - 'paths/data-science'
+---
+
+**Stochastic Gradient Descent** (SGD) is a optimization algorithm. It is variant of gradient descent optimizer. The SGD minimize the loss function of machine learning algorithms and deep learning algorithms during backpropagation to update the weight and bias in Artificial Neural Networks. 
+
+The term stochastic mean randomness on which algorithm based upon. In this algorithm instead of taking whole dataset like grdient descent we take single randomly selected data point or small batch of data.suppose if the data set contains 500 rows SGD update the model parameters 500 times in one cycle or one epoch.
+
+This approach significantly reduces computation time, especially for large datasets, making SGD faster and more scalable.SGD is used for training models like neural networks, support vector machines (SVMs), and logistic regression. However, it introduces more noise into the learning process, which can lead to less stable convergence but also helps escape local minima, making it suitable for non-convex problems. 
+
+
+![stochastic gradient descent](https://www.goglides.dev/images/Jq8EpuPoMjCcxm7PqMqWuQK7M_MoVtdfAUsGJsoUIMA/w:880/mb:500000/ar:1/aHR0cHM6Ly93d3ct/Z29nbGlkZXMtZGV2/LnMzLmFtYXpvbmF3/cy5jb20vdXBsb2Fk/cy9hcnRpY2xlcy8z/cGh3bjR0bmpnNGlo/eHV0Znpqby5wbmc)
+
+## Algorithms Step
+
+- At each iteration, a random sample  is selected from the training dataset.
+- The gradient of the cost function with respect to the model parameters is computed based on the selected sample.
+- The model parameters are updated using the computed gradient and the learning rate.
+- The process is repeated for multiple iterations until convergence or a specified number of epochs.
+
+## Formula 
+
+$$ 
+\large \theta = \theta - \alpha  * \nabla J((\theta ; x_iy_i))
+$$
```

Comment:
```suggestion
$$ 
\large \theta = \theta - \alpha  \cdot  \nabla J(\theta ; x_i, y_i)
$$
``` 
Typically \cdot is used for multiplication to keep the LaTex notation standard and  the ` * ` symbol is not used.